### PR TITLE
WIP GraphPanel: use declarative options setup

### PR DESCRIPTION
**Pull request by ryantxu**
This is an attempt to have the graph panel use some of the fancy new settings -- it will be some work ;)

### CODE REVIEW

File: public/app/features/dashboard/state/PanelModel.ts [hunk 1]
```diff
@@ -310,7 +310,7 @@ export class PanelModel implements DataConfigSource {
   pluginLoaded(plugin: PanelPlugin) {
     this.plugin = plugin;
 
-    if (plugin.panel && plugin.onPanelMigration) {
+    if (plugin.onPanelMigration) {
```
Comment (ryantxu) [inline public/app/features/dashboard/state/PanelModel.ts:313]: this lets anyone with a migrationHandler get called... not just react panels

File: public/app/plugins/panel/graph/module.ts [hunk 2]
```diff
@@ -351,4 +354,81 @@ class GraphCtrl extends MetricsPanelCtrl {
...
+      .addBooleanSwitch({
+        category: 'Draw Mode',
+        path: 'points',
+        name: 'Points',
+        description: 'show bars',
+        defaultValue: false,
+      })
+      .addBooleanSwitch({
+        category: 'Legend',
+        path: 'legend.isVisible',
```
Comment (ryantxu) [inline public/app/plugins/panel/graph/module.ts:385]: @dprokop -- are nested values supported?  or does this have to be all at the root level?

**Comment by ryantxu** [comment]
@dprokop -- can you take a look?  in particular what is the pattern for categories?  This made:
image

and has a bunch of console errors!

**Comment by torkelo** [comment]
Nice was thinking of experimenting with this as well :) 

The category property is an array of strings (to supported nested categories)

File: public/app/plugins/panel/graph/module.ts (hunk 2 above)
Comment (torkelo) [reply public/app/plugins/panel/graph/module.ts:385]: yes, this should be supported

Comment (dprokop) [reply public/app/plugins/panel/graph/module.ts:385]: yes they are!

**Comment by dprokop** [comment]
> @dprokop -- can you take a look? in particular what is the pattern for categories? This made:
> image
> 
> and has a bunch of console errors!

Categories is an array of strings. Added some more docs love to these properties in this PR: https://github.com/grafana/grafana/pull/23495

File: public/app/plugins/panel/graph/module.ts [hunk 3]
```diff
@@ -351,4 +354,81 @@ class GraphCtrl extends MetricsPanelCtrl {
...
+        description: 'show bars',
+        defaultValue: true,
+      })
+      .addBooleanSwitch({
+        category: ['Draw Mode'],
+        path: 'points',
+        name: 'Points',
+        description: 'show bars',
+        defaultValue: false,
+      })
```
Comment (torkelo) [inline public/app/plugins/panel/graph/module.ts:382]: these exist as series overrides as well so technically should be field config level options

**Pull request closed without merging**